java/src/main/java/com/logicalclocks/hsfs/engine/FeatureGroupEngine.java

```diff
@@ -187,27 +188,46 @@ private void saveOnlineDataframe(FeatureGroup featureGroup, Dataset<Row> dataset
     SparkEngine.getInstance().writeOnlineDataframe(dataset, saveMode, writeOptions);
   }
 
-  public Map<String, Map<String, String>> commitDetails(FeatureGroup featureGroup, Integer limit)
-      throws IOException, FeatureStoreException {
-    List<FeatureGroupCommit> featureGroupCommits = featureGroupApi.commitDetails(featureGroup, limit);
+  private Map<Long, Map<String, String>>  getCommitDetails(FeatureGroup featureGroup, String wallclockTime,
+                                                           Integer limit)
+      throws FeatureStoreException, IOException, ParseException {
+
+    Long wallclockTimestamp =  wallclockTime != null ? utils.getTimeStampFromDateString(wallclockTime) : null;
+    List<FeatureGroupCommit> featureGroupCommits =
+        featureGroupApi.getCommitDetails(featureGroup, wallclockTimestamp, limit);
     if (featureGroupCommits == null) {
       throw new FeatureStoreException("There are no commit details available for this Feature group");
     }
-    Map<String, Map<String, String>> commitDetails = new HashMap<String, Map<String, String>>();
+    Map<Long, Map<String, String>> commitDetails = new HashMap<>();
     for (FeatureGroupCommit featureGroupCommit : featureGroupCommits) {
-      commitDetails.put(featureGroupCommit.getCommitID().toString(), new HashMap<String, String>() {{
+      commitDetails.put(featureGroupCommit.getCommitID(), new HashMap<String, String>() {{
```

Comment:
why are the values stored as strings, aren't they all integers?

Reply:
We return `commitedOn` as human readable date string. Then to avoid returning as `Object` I convert other values as `String` also